File: frontend/src/types/asset.ts
export type AssetType = 'pdf' | 'txt' | 'doc' | 'note' | 'data' | 'code' | 'result';

export type Asset = {
  id: number;
  workflow_id: number;
  name: string;
  type: AssetType;
  file_path: string;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
};

export type AssetStructurerMetadata = {
  skill_name: string;
  version: string;
  status: string;
  summary: string;
  structured_at: string | null;
  error?: string | null;
};

export type AssetUploadInput = {
  workflowId: number;
  type: AssetType;
  name?: string;
  metadataJson?: Record<string, unknown> | null;
  file: File;
};

File: backend/app/schemas/asset.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


AssetType = Literal["pdf", "txt", "doc", "note", "data", "code", "result"]


class AssetCreate(BaseModel):
    workflow_id: int
    name: str = Field(..., max_length=255)
    type: AssetType
    file_path: str
    metadata_json: dict[str, Any] | None = None


class AssetRead(ORMModel):
    id: int
    workflow_id: int
    name: str
    type: str
    file_path: str
    metadata_json: dict[str, Any] | None = None
    created_at: datetime

